src/xtgeo/grid3d/_gridprop_value_init.py

```diff
@@ -1,101 +1,141 @@
 """GridProperty (not GridProperies) some etc functions"""
+from __future__ import annotations
 
-# from ._gridprop_import_eclrun import import_eclbinary
-#  --> INIT, UNRST
-#
-# from ._gridprop_import_grdecl import import_grdecl_prop, import_bgrdecl_prop
-#  --> ASCII and BINARY GRDECL format
-# from ._gridprop_import_roff import import_roff
-#  --> BINARY ROFF format
-
+from typing import TYPE_CHECKING, Optional, Tuple, Union
 
 import numpy as np
 
 import xtgeo
+from xtgeo.common import XTGeoDialog
 
-xtg = xtgeo.common.XTGeoDialog()
+logger = XTGeoDialog().functionlogger(__name__)
 
-logger = xtg.functionlogger(__name__)
+if TYPE_CHECKING:
+    from xtgeo.grid3d import Grid, GridProperty
 
 
-def gridproperty_non_dummy_values(gridlike, dimensions, values, isdiscrete):
+def gridproperty_non_dummy_values(
+    gridlike: Optional[Union[Grid, GridProperty]],
+    dimensions: Tuple[int, int, int],
+    values: Optional[Union[np.ndarray, float, int]],
+    isdiscrete: bool,
+) -> np.ma.MaskedArray:
     """Gives the initial values array of an gridprop.
 
-    Param:
-      gridlike: Either Grid or GridProperty, giving the mask to replicate.
-      dimensions: The dimensions of the gridprop
-      values: The values parameter given to init.
-      isdiscrete: The discrete parameter given to init.
+    Parameters:
+        gridlike:
+            Either Grid or GridProperty, giving the mask to replicate.
+        dimensions:
+            The (ncol, nrow, nlay) dimensions of the grid property.
+        values:
+            The values parameter given to init.
+        isdiscrete:
+            The discrete parameter given to init.
 
     Returns:
-      The array to be set to GridProp._values.
+        The array to be set to GridProperty._values.
+
     """
     if values is None:
-        values = initial_gridprop_values_zero(dimensions, isdiscrete)
-    elif np.isscalar(values):
-        values = initial_gridprop_values_from_scalar(dimensions, values, isdiscrete)
+        _values = initial_gridprop_values_zero(dimensions, isdiscrete)
+    elif isinstance(values, (int, float)):
```

Comment:
Might be somewhat of an edgecase:
```python
In [1]: from fractions import Fraction

In [2]: isinstance(Fraction(5, 16), (float, int))
Out[2]: False
````

Reply:
Nice catch, I wouldn't have thought of this. Changed it to `isinstance(values, numbers.Number)` which I think hits the specificity intended here (`np.isscalar` is permissive enough to allow strings)